pandas/tests/io/excel/test_readers.py

```diff
@@ -508,10 +508,10 @@ def test_reader_spaces(self, read_ext):
     def test_read_excel_ods_nested_xml(self, read_ext, basename, expected):
         # see gh-35802
         engine = pd.read_excel.keywords["engine"]
-        if engine != "odf":
+        if engine not in ("odf", None):
             pytest.skip(f"Skipped for engine: {engine}")
 
-        actual = pd.read_excel(basename + read_ext)
+        actual = pd.read_excel(basename + ".ods")
```

Comment:
We can now run these tests on the `None` engine since `read_excel` will now correctly infer the `odf` engine rather than falling through to `xlrd`.